### ISSUE TITLE
[Question] Intervalos exagerados após otimizar

### ISSUE THREAD

**Issue by JeffersonCorreia32a** [open-spaced-repetition/fsrs4anki#389]
**Research**

*Enter an [x] character to confirm the points below:*

- [x ] I have checked the [FAQ](https://github.com/open-spaced-repetition/fsrs4anki/wiki/FAQ) and could not find an answer to my question
- [x ] I have read the [wiki](https://github.com/open-spaced-repetition/fsrs4anki/wiki) and still felt confused
- [x ] I have searched for similar existing questions here

**Question**

Dear all,

I would like to understand why my deck is displaying such absurd values. The optimizer is providing parameters that deviate significantly from the original values. When I optimize the deck, the "w" values increase to an unreasonable extent, causing the application of these values to result in absurdly long intervals between card reviews. I don't understand the underlying problem.

The parameters I used were as follows:

Next Day Starts at (Step 2): 12
Timezone (Step 3.1): America/Bahia
Filter out suspended cards: checked
Advanced Settings (Step 3.2): 0.9
Revlog Start Date: Optimize review logs after this date: 2006-10-05
Example original "w" values: If I keep the original "w" as follows:
[0.4, 0.6, 2.4, 5.8, 4.93, 0.94, 0.86, 0.01, 1.49, 0.14, 0.94, 2.18, 0.05, 0.34, 1.26, 0.29, 2.61]
A new card will have a good interval of 2 days and an easy interval of 7 days.

Example optimized "w" values: If I use the optimized "w" for my deck as follows:
[1.46, 4.9, 8.82, 10.08, 4.8994, 0.8604, 0.8373, 0.007, 1.5252, 0.1, 0.9738, 2.2595, 0.0204, 0.3773, 1.5048, 0.3335, 2.3037]
A new card will have a good interval of 10 days and an easy interval of 12 days.

A good interval of 10 days for a new card seems clearly absurd...

I would like to know if others are experiencing such significant deviations from the original parameters or if it's just affecting my deck. This might help identify the issue.

This problem has been occurring for me since version 3, but it seems to have worsened recently.

I believe it would be helpful to find a way to maintain these parameters within an acceptable standard deviation for a human. In my opinion, it's unrealistic for a human to fully memorize content from today to 10 days in the future after seeing it only once. Perhaps, if the optimizer could set a maximum good interval of 3 or 4 days within a standard deviation, it would be more reasonable.

These discrepancies persist in older reviews, resulting in significantly increased intervals. Additionally, there is a substantial difference between the "good" and "easy" buttons. For example, one card had a good interval of 29 days while the easy interval was 2.6 months, representing a 250% difference.

I have included my complete deck, including media, for your evaluation. Thank you in advance for any effort made to help solve this case.

[0PROJETO CONCURSO 20mil+26-07-2023 com midia.zip](https://github.com/open-spaced-repetition/fsrs4anki/files/12171189/0PROJETO.CONCURSO.20mil%2B26-07-2023.com.midia.zip)

**Comment by L-M-Sherlock** [comment]
> A good interval of 10 days for a new card seems clearly absurd...

It doesn't. Some users have 30+ days for a new card when they press easy.

I checked your collection. The first forgetting curve when you press easy is: 

![image](https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/07ccc865-31cd-43a5-9f1c-edbb23e999ac)

This graph shows that your retention is higher than 90% even when the interval is 9 days.

**Comment by JeffersonCorreia32a** [comment]
> > A good interval of 10 days for a new card seems clearly absurd...
> 
> It doesn't. Some users have 30+ days for a new card when they press easy.
> 
> I checked your collection. The first forgetting curve when you press good is:
> 
> ![image](https://user-images.githubusercontent.com/32575846/256233031-5293f0c5-d514-4ecb-aeb1-80f4db3b7c14.png)
> 
> This graph shows that your retention is near 98% in 5 days.
> 
> The first forgetting curve when you press easy is:
> 
> ![image](https://user-images.githubusercontent.com/32575846/256231038-c1934f16-2575-4640-9e7d-0fc97c24a991.png)
> 
> This graph shows that your retention is higher than 90% even when the interval is 9 days.

I really don't think it's reasonable to have a 30-day interval for the first review, out of two 1, either this parameter called retention is hooked or else the cards are too easy. Formerly, there was talk of the first review with 24 hours and the second with 7 days. Jumping from 1 review in 24 hours to 10 days is a lot in my opinion.

My cards are very heterogeneous and there are very difficult and very easy cards.

I'm going to try to select the cards that I find easy and suspend and redo the optimization only with cards that I consider medium or difficult, because I don't want to have these cards harmed by the easy ones, I'd rather waste time and really have the expected retention, than have a false sense of that I am withholding some of the content.


I would like the opinion of several users on this issue.

**Comment by L-M-Sherlock** [comment]
> I really don't think a 30-day interval for the first review is reasonable, one of the two, either this parameter called retention is addicted or the cards are too easy.

![image](https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/3188f84b-23f9-46fd-9db7-0790ce31d3a8)

This forgetting curve is from my collection.

![image](https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/3cb6cc4b-bc46-411b-bc4b-4445178e1b33)

This forgetting curve is from @Expertium. His initial stability for easy is 200+ days.

**Comment by user1823** [comment]
@JeffersonCorreia32a, the initial stability for Good (the third value in w) in my collection is 13 days. Initially, I also thought that it was too large.

But, later while analyzing the data, I found that I forget only about 2% of the cards if I review them after 2-4 days. So, it makes sense that the stability would be around 13 days. (Remember that stability is defined as the time when the retrievability falls to 90%.)

Not only that, I could quite comfortably recall most of my cards when they come up for review. So, FSRS is right (at least for my collection) in saying that the first review interval should not be that small as users usually think. Also, don't think that I am learning something that is too easy.

A suggestion: The requestRetention option in the scheduler code is there to control how much fraction of the cards you retain when FSRS shows the cards to you. So, if you think that you are forgetting too much, you might want to increase the requestRetention.

**Comment by Expertium** [comment]
> This forgetting curve is from @Expertium. His initial stability for easy is 200+ days.

In my case, that's because I already know some material that I have encountered outside of Anki.

**Comment by galantra** [comment]
More reasons:

1. > [Do not neglect the basics. Memorizing seemingly obvious things is not a waste of time!](https://www.supermemo.com/en/blog/twenty-rules-of-formulating-knowledge)
2. A deck was created by someone else (or even by a program). E.g., a comprehensive language deck or Ultimate Geography.
3. Generous use of cloze deletion (especially now that [nested clozes are possible](https://github.com/ankitects/anki/pull/2141)).
4. [Transfer of learning](https://en.wikipedia.org/wiki/Transfer_of_learning).

After all, it's one of the goals of the [20 rules of formulating knowledge](https://www.supermemo.com/en/blog/twenty-rules-of-formulating-knowledge) to make cards _easy_.

**Comment by JeffersonCorreia32a** [comment]
Alright, I'll go ahead and test doing the first review with the recommended time from the optimized "w." By the way, could you share the learning steps you use? I've been using only 20m and nothing else, but I've been considering trying 20m, 1d, since the first review scheduled by FSRS is 8 to 10 days, and the maximum recommended for the learning step is 1d. Additionally, I have a feeling that this 24-hour review is essential. Please let me know what you think and what learning steps you use.

**Comment by AnyonicFugue** [comment]
I also encountered the problem of exaggerated ranges, but much more severe. Here is the export of my learning data: 
[Anki Data.zip](https://github.com/open-spaced-repetition/fsrs4anki/files/12193933/Anki.Data.zip)

The V4 optimizer tells me the optimal parameters are
`[23.47, 30.0, 22.37, 23.47, 4.4425, 0.5031, 0.6348, 0.2575, 1.9887, 0.1041, 1.4385, 2.6968, 0.01, 0.8163, 1.7731, 0.2265, 2.9595]`
which gives insane intervals like:
```
rating history: 2,3,3,1,3,1,1,3,3,3,3,3
interval history: 0,30,177,910,822,3457,2445,1840,6584,22653,36500,36500,36500
difficulty history: 0,4.9,4.8,4.7,5.6,5.3,6.0,6.6,6.0,5.6,5.3,5.1,4.9
```

When I try the V3 optimizer, the parameters seem much more sensible:
`[1.4617, 1.4644, 4.5713, -0.1261, -0.3351, 0.4935, 1.8344, -0.15, 1.2098, 2.3701, -0.01, 0.5964, 1.3038]`
with intervals
```
rating history: 2,3,3,1,3,1,1,3,3,3,3,3
interval history: 0,0,3,16,14,61,31,21,87,312,987,2793,7178
difficulty history: 0,4.7,4.6,4.6,4.9,4.8,5.0,5.1,4.9,4.7,4.6,4.6,4.6
```

Some other information that might be useful:
I set `requestedRetention = 0.9`.
Currently schedule I use in Anki: ![image](https://github.com/open-spaced-repetition/fsrs4anki/assets/118508626/e3d19b97-6496-4ba0-b064-09eb5ea2b97c)

**Comment by L-M-Sherlock** [comment]
> I also encountered the problem of exaggerated ranges, but much more severe. Here is the export of my learning data:
> [Anki Data.zip](https://github.com/open-spaced-repetition/fsrs4anki/files/12193933/Anki.Data.zip)

<img width="871" alt="image" src="https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/01022682-aab3-409c-995c-ca5feece88b1">

<img width="871" alt="image" src="https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/01022682-aab3-409c-995c-ca5feece88b1">

I found that your forgetting curves are weird.

**Comment by AnyonicFugue** [comment]
> > I also encountered the problem of exaggerated ranges, but much more severe. Here is the export of my learning data:
> > [Anki Data.zip](https://github.com/open-spaced-repetition/fsrs4anki/files/12193933/Anki.Data.zip)
> 
> I found that your forgetting curves are weird. Your retention is very high in first 10 days.

I think the strange forgetting curves might be from my previous (unreasonable) schedule, i.e. `Learning Steps = 3d, 12d, 21d, ...` (I forgot the exact values but they are not too far away). When using that schedule my retention is much lower than now. This might explain the low retention rates at certain values of intervals.

Also, I reviewed very few cards for a week in march when cards are piling up, which led to lots of forgetting. I guess that also contributed to my strange forgetting curves.

**Comment by AnyonicFugue** [comment]
So I wonder is there a way to produce reasonable fsrs parameters. Should I delete the data points with exceptionally low retention and run the optimizer again?

Additional information that might be useful: I'm mainly using Anki to memorize English words to prepare for the GRE exam. The difficulty of different words can be very different for me.
My memory is, I think, a bit above average, but far below the V4 optimizer predicts.

**Comment by Expertium** [comment]
> Shouldn't the longest gap of the learning steps be Max 1d? Then I'm going to ask: what is the ideal learning steps configuration to use with FSRS4?

When you are using FSRS, yes, the longest learning step should be 1d. But I'm assuming @AnyonicFugue was using the default Anki scheduler with his 3d, 12d, 21d steps.
The really weird part is the fact that initial stabilities seem to be about the same for all grades:

> The V4 optimizer tells me the optimal parameters are
> `[**23.47, 30.0, 22.37, 23.47**, 4.4425, 0.5031, 0.6348, 0.2575, 1.9887, 0.1041, 1.4385, 2.6968, 0.01, 0.8163, 1.7731, 0.2265, 2.9595]`

The first 4 numbers are initial values of S for Again/Hard/Good/Easy. And they all are about the same. I've never seen anything like this, usually the values monotonically increase, like 1, 2, 4, 8, for example. Unless @AnyonicFugue just randomly chooses which button to press with his eyes closed, I don't know how to explain this.

**Comment by JeffersonCorreia32a** [comment]
> > O intervalo mais longo das etapas de aprendizado não deveria ser Max 1d? Então eu vou perguntar: qual é a configuração de etapas de aprendizado ideal para usar com o FSRS4?
> 
> Quando você estiver usando FSRS, sim, a etapa de aprendizado mais longa deve ser 1d. Mas eu estou assumindo@AnyonicFugueestava usando o agendador Anki padrão com suas etapas 3d, 12d, 21d. A parte realmente estranha é o fato de que as estabilidades iniciais parecem ser as mesmas para todas as séries:
> 
> > O otimizador V4 me diz que os parâmetros ideais são 
> > [ **23.47, 30.0, 22.37, 23.47** , 4.4425, 0.5031, 0.6348, 0.2575, 1.9887, 0.1041, 1.4385, 2.6968, 0.01, 0.8163, 1.7 731, 0,2265, 2,9595]
> 
> Os primeiros 4 números são valores iniciais de S para Novamente/Difícil/Bom/Fácil. E todos eles são quase iguais. Nunca vi nada assim, geralmente os valores aumentam monotonicamente, como 1, 2, 4, 8, por exemplo. A menos que@AnyonicFugueapenas escolhe aleatoriamente qual botão apertar com os olhos fechados, não sei como explicar isso.

Again I beg for information on proper learning steps. I'm only using one step at the moment. 20m and the good one is already determined by FSRS4. I don't know if I'm doing it right. or if for example there would have to be more learning steps like for example: 5m 20m 120m 1d.

This strikes me as a huge information gap that I need to fill.

**Comment by Expertium** [comment]
> What are the most recommended learning steps to use with FSRS4?

1d or shorter.

**Comment by L-M-Sherlock** [comment]
<img width="791" alt="image" src="https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/3fdd0b2a-c693-47e3-b35c-7a949ae6e869">

I think there are two completely different types of materials.

**Comment by L-M-Sherlock** [comment]
<img width="800" alt="image" src="https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/3d0b2e8c-a131-453a-b7bc-77e99658b3b8">

<img width="800" alt="image" src="https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/bd5ccebd-057f-4b4b-86e6-44d6d6e4e633">

**Comment by JeffersonCorreia32a** [comment]
> > What are the most recommended learning steps to use with FSRS4?
> 
> 1d or shorter.

Ok mas quantos passos e com quanto tempo cada exatamente, é isso que preciso saber. Afinal existe uma infinidade de possibilidades que podem prejudicar a eficiência e até invalidar toda a eficiência do FSRS4, se essa escolha fica aberta pra cada um escolher podem oxorrer absurdos, por exemplo:
Exemplo absurdo : 1m 5m 10m 15 m 30m 60m 120m 240m 480m 1d
O exemplo acima faria com que a pessoa perdesse muito tempo clicando caso clicasse em bom até passar pra o agendamento pelo FSRS4. Isso não é um learning steps eficiente.

Exemplo 2 que uso hoje: 20m
Esse exemplo é o outro extremo que utilizo hoje, pois o 20m gica no botao(errei) e o bom já é determinado pelo FSS4. Entendo que deveria ter uma diretriz mais clara do que é mais recomendado.

**Comment by Expertium** [comment]
@JeffersonCorreia32a I always use 2 steps: 15m 1d. I don't recommend using a lot of small steps, like 5m 10m 15m 30m 60m, as it has a negligible impact on long-term learning.

**Comment by nb9618** [comment]
I might be mistaken, but as I understand it, the optimizer discards all the learning (or relearning) stage reviews except the first ones. This means that the optimizer does not consider any reviews that have intervals longer than a day, and their impact on stability and difficulty. What I think is that these reviews, while technically being "learning" type reviews, are not any different from ordinary "review" type reviews in terms of their relation to the memory model. It may be beneficial for the optimizer instead of throwing away all the reviews in a single "learning" session, split the session by days, and keep the first review of each day. I think this should  produce better results for users who happen to have long learning intervals in their review histories.

**Comment by nb9618** [comment]
In other words, the optimizer should pick every first review of the day, regardless of its type
![Untitled](https://github.com/open-spaced-repetition/fsrs4anki/assets/137687989/ab4aec79-6cc3-461a-a54e-6e9bd724926a)
(Image taken from https://www.reddit.com/r/Anki/comments/jl9w17/cards_i_relearned_yesterday_which_should_have/)

**Comment by L-M-Sherlock** [comment]
> I might be mistaken, but as I understand it, the optimizer discards all the learning (or relearning) reviews except the first ones in each session. This means that the optimizer does not consider any reviews that have intervals longer than a day, and their impact on stability and difficulty. What I think is that these reviews, while technically being "Learn" type reviews, are not any different from ordinary "Review" type reviews in terms of their relation to the memory model. It may be beneficial for the optimizer instead of throwing away all the reviews in a single "Learn" or "Relearn" session, split the session by days, and keep the first review of each day. I think this should produce better results for users who happen to have long learning intervals in their review histories.

The optimizer doesn't discards all the learning (or relearning) reviews except the first ones in each session. The optimizer has split the session by days and kept the first review of each day. Please check the code here: 

https://github.com/open-spaced-repetition/fsrs-optimizer/blob/420a10f6c1daad0bdef3c59fe8b184598a6a0761/src/fsrs_optimizer/fsrs_optimizer.py#L383

**Comment by nb9618** [comment]
Unfortunately, processing the raw data is the part that I struggle to understand. If I was wrong, then I'm sorry for bothering.

**Comment by AkiraChisaka** [comment]
I want to chime in, and add that I am also having similar experiences.

With the v4 algorithm, it seems like it is a bit too aggressive and giving out huge `w3` numbers.

For me, my w is [0.13, 0.18, 2.15, 30.0, 4.9269, 0.9457, 0.8685, 0.0056, 1.3993, 0.1, 0.8463, 2.2054, 0.0263, 0.3629, 1.2524, 0.3426, 2.6167]. I have ran the v4 optimizer quite a few times in the past few days, and it always gives me `30.0`, which I believe is the max value for `w3`?

Considering that, I have been using Anki for exactly one month, and FSRS for no more than 3 weeks, isn't the number `30.0` just way too high? I am very uncertain if my retention for a card I rated as "easy" 30 days later will be anything close to 90%. Since I'm pretty sure the retention degradation is not 100% linear.

So yeah, I want to propose, that there should a cap on the length of the first interval after pressing "easy" on a card. Something like, cap it to how long you have been using Anki. This way, at least I know that FSRS's initial "easy" interval is somewhat based on precedence.

And yeah, I think in general, v4 feels a bit too... sensitive and aggressive. As in, when there isn't that much training data for the optimizer, as a user, it feels a bit like... FSRS is "just winging it" a lot of the times. So I feel like FSRS v4 should be a bit more conservative when deciding parameters, mostly for new users that are new to Anki in general and don't actually have that much review data.

Also, I think my requested retention is 90%, but my actual ratio of "none again" answered for young cards is only 80.37%. And I do admit it's starting to feel a bit demoralizing, feeling like the algorithm keeps giving me cards that I just forgotten.

Also here is all my data, feel free to check it out if it's relevant: [collection-2023-07-29@16-31-40.zip](https://github.com/open-spaced-repetition/fsrs4anki/files/12206668/collection-2023-07-29%4016-31-40.zip)

**Comment by L-M-Sherlock** [comment]
> I want to chime in, and add that I am also having similar experiences.

Could you set `verbose=True` for `optimizer.pretrain(verbose=False)` in the section 2.1 when you run the optimizer? We can check your forgetting curve for first easy rating.

**Comment by user1823** [comment]
@L-M-Sherlock and @Expertium, I am seeing that the highest number of support requests that we are getting these are related to high values of initial stability, especially from users who don't have sufficient review data.

This new approach of estimating the initial stability worked well for us, who have large amount of data. But, I think that it is not well suited for users who have insufficient review data.

So, I think that we should set a threshold below which the optimizer is not allowed to use this approach.

For example, we can use a total of 2000 cards as the minimum requirement to use curve_fit. If the total number of cards is less than 2000, we can bring back v3's pre-train approach (but save the 4 parameters for initial stability in the final w instead of 2, to retain compatibility with the v4 code).

The exact thresholds for deciding whether to use v4 approach, or to use v3 approach or to just say that the data is insufficient would need to be decided based on experiments with data of people having low number of reviews and also small samples from our own collections.

Note that there is a significant difference between the two types of datasets that I suggested to use:
- In collections from new users of Anki, both the number of cards and the number of reviews per card would be low.
- In small samples from our own collections, the number of cards would be low, but the number of reviews per card would be high.

**Comment by JeffersonCorreia32a** [comment]
> @L-M-Sherlocke@Expertium, estou vendo que o maior número de solicitações de suporte que estamos recebendo está relacionado a altos valores de estabilidade inicial, especialmente de usuários que não possuem dados de revisão suficientes.
> 
> Essa nova abordagem de estimativa da estabilidade inicial funcionou bem para nós, que temos uma grande quantidade de dados. Mas acho que não é adequado para usuários com dados de revisão insuficientes.
> 
> Portanto, acho que devemos definir um limite abaixo do qual o otimizador não pode usar essa abordagem.
> 
> Por exemplo, podemos usar um total de 2.000 cartões como requisito mínimo para usar curve_fit. Se o número total de cartões for menor que 2.000, podemos trazer de volta a abordagem pré-treinamento do v3 (mas salve os 4 parâmetros para estabilidade inicial no w final em vez de 2, para manter a compatibilidade com o código v4).
> 
> Os limites exatos para decidir se usar a abordagem v4, ou usar a abordagem v3 ou apenas dizer que os dados são insuficientes precisariam ser decididos com base em experimentos com dados de pessoas com baixo número de avaliações e também pequenas amostras de nossas próprias coleções .
> 
> Observe que há uma diferença significativa entre os dois tipos de conjuntos de dados que sugeri usar:
> 
> * Nas coletas de novos usuários do Anki, tanto o número de cartões quanto o número de avaliações por cartão seriam baixos.
> * Em pequenas amostras de nossas próprias coleções, o número de cartões seria baixo, mas o número de revisões por cartão seria alto.

Sugiro outra coisa ao em vez de limitar a um numero minimo de 2000 mil cartões por exemplo a função completa do otimizador. Sugiro que ao ser calculado a otimização os valores de w seja limitado a um desvio padrão gradativamente, de forma que quanto mais cartões e revlogs tiver mais livre desse desvio padrão ficará.  Segue imagem para ilustrar a ideia:
![image](https://github.com/open-spaced-repetition/fsrs4anki/assets/138017463/f2ddb185-4601-485d-b12c-877a30463369)

**Comment by Expertium** [comment]
That's an interesting idea, though I don't know how to choose the values of "bands".

**Comment by L-M-Sherlock** [comment]
> Also here is all my data, feel free to check it out if it's relevant: [collection-2023-07-29@16-31-40.zip](https://github.com/open-spaced-repetition/fsrs4anki/files/12206668/collection-2023-07-29%4016-31-40.zip)

I test your collection. I find that the problem is not caused by 

```python
            params, _ = curve_fit(power_forgetting_curve, delta_t, recall, sigma=1/np.sqrt(count), bounds=((0.1), (30 if total_count < 1000 else 365)))
```

It is caused by

```python
        def S0_rating_curve(rating, a, b, c):
            return np.exp(a + b * rating) + c

        params, covs = curve_fit(S0_rating_curve, list(rating_stability.keys()), list(rating_stability.values()), sigma=1/np.sqrt(list(rating_count.values())), method='dogbox', bounds=((-15, 0.03, -5), (15, 7, 30)))
```

Without the limit of 100 reviews, the fitting results are:

![pretrain_0](https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/0a0c7e4e-09aa-4ffc-8caf-bf9bc6046030)

![pretrain_1](https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/68e268ff-e27a-4bbe-ba70-92438ecb4788)

![pretrain_2](https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/2f13f716-b05a-4bd1-84ff-e30cb80f9b19)

![pretrain_3](https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/305c8b4f-4162-4b9f-b8a2-2fa5d75f1111)

With the limit of 100 reviews, the fitting results are: 

![pretrain_0](https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/17a13d8c-999e-41df-b84e-25a46bd20821)

![pretrain_1](https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/80581ee3-b1c2-4618-97bc-fc9ede49a9c8)

![pretrain_2](https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/4b02944e-24e3-4816-ad80-14ca81edbc75)

**Comment by L-M-Sherlock** [comment]
> I suggest something else instead of limiting the full function of the optimizer to a minimum number of 2000 thousand cards for example. I suggest that, when calculating the optimization of the values ​​of w, it is gradually limited to one standard deviation, so that the more cards and revlogs there are, the freer they are from this expected standard deviation with increasing bands. Here is an image to illustrate the idea:

I have an idea. We can set `p0` (the initial guess) and `maxfev` (the number of iteration) for the curve_fit method. A small `maxfev` could decrease the deviation between the fitting result and the initial guess.

**Comment by user1823** [comment]
> I have an idea. We can set `p0` (the initial guess) and `maxfev` (the number of iteration) for the curve_fit method. A small `maxfev` could decrease the deviation between the fitting result and the initial guess.

By curve_fit, do you mean the following?
```py
params, _ = curve_fit(power_forgetting_curve, delta_t, recall, sigma=1/np.sqrt(count), bounds=((0.1), (30 if total_count < 1000 else 365)))
```

If yes, I like the idea. But, we will have to test it with a wide range of collections (ones with large number of reviews as well as ones with small number of reviews) to ensure that it doesn't produce unintended results.
